app/src/main/java/android/tungpt/lesson6/MainActivity.java

```diff
@@ -1,13 +1,102 @@
+import android.tungpt.lesson6.Adapter.ContactAdapter;
+import android.tungpt.lesson6.Model.Contact;
+
+import java.util.ArrayList;
 
 public class MainActivity extends AppCompatActivity {
+    public static final Uri CONTENT_URI = ContactsContract.Contacts.CONTENT_URI;
+    public static final String WHERE = " = ? ";
+    private static final int REQUEST_PERMISSION = 1;
+    private ContentResolver mContentResolver;
+    private ArrayList<Contact> mContacts;
+    private ContactAdapter mContactAdapter;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        mContacts = new ArrayList<>();
+        checkPermission();
+    }
+
+    public void initView() {
+        RecyclerView recyclerView = findViewById(R.id.recycler_view);
+        recyclerView.setHasFixedSize(true);
+        mContactAdapter = new ContactAdapter(this, mContacts);
+        recyclerView.setAdapter(mContactAdapter);
+        loadContacts();
+    }
+
+    public ArrayList<Contact> loadContacts() {
+        ArrayList<Contact> contacts = new ArrayList<>();
+        mContentResolver = getContentResolver();
+        Cursor cursor = mContentResolver.query
+                (CONTENT_URI, null, null, null, null);
+        if (cursor != null && cursor.getCount() > 0) {
+            while (cursor != null && cursor.moveToNext()) {
+                String id = cursor.getString(cursor.getColumnIndex(ContactsContract.Contacts._ID));
+                String name = cursor.getString(cursor.getColumnIndex
+                        (ContactsContract.Contacts.DISPLAY_NAME));
+                int phoneNumber = Integer.parseInt(cursor.getString(cursor.getColumnIndex
+                                (ContactsContract.Contacts.HAS_PHONE_NUMBER)));
+                if (phoneNumber > 0) {
+                    Cursor phoneCursor = mContentResolver.query(
+                            ContactsContract.CommonDataKinds.Phone.CONTENT_URI,
+                            null,
+                            ContactsContract.CommonDataKinds.Phone.CONTACT_ID + WHERE,
+                            new String[]{id},
+                            null);
+                    while (phoneCursor != null && phoneCursor.moveToNext()) {
+                        String number = phoneCursor.getString(phoneCursor.getColumnIndex
+                                (ContactsContract.CommonDataKinds.Phone.NUMBER));
+                        mContacts.add(new Contact(name, number));
+                        mContactAdapter.notifyDataSetChanged();
+                    }
+                    phoneCursor.close();
+                }
+            }
+            cursor.close();
+        }
+        return contacts;
+    }
+
+    private void checkPermission() {
+        String[] permissions = {Manifest.permission.READ_CONTACTS};
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
```

Comment:
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
                && ContextCompat.checkSelfPermission(this, permissions[0])
                != PackageManager.PERMISSION_GRANTED) {
            requestPermissions(permissions, REQUEST_PERMISSION);
            return;
        } 
         initView();